File: 5/2. Линейный поиск/G. Ни больше ни меньше/mike.py
from tests.index import test, input
test(0)

__input__ = input
def input():
  try: return __input__()
  except: return None


for _ in range(int(input())):
  n = int(input())
  a = list(map(int, input().split()))
  
  ans, prevbreak, curmin = [], 0, n
  
  
  for i in range(n):
    curmin = min(curmin, a[i])

    if curmin > i - prevbreak: continue
      
    ans.append(i - prevbreak)
    prevbreak = i
    curmin = a[i]
    
  
  ans.append(n - prevbreak)
  print(len(ans))
  print(*ans)

File: 6/4. Деревья, представления и обходы/H. Вершинно-реберное покрытие дерева/mike.js
const fs = require('fs')
const test = 2
const __input__ = fs.readFileSync(`${__dirname}/tests/${test}`, 'utf8').split('\n')
const answer = fs.readFileSync(`${__dirname}/tests/answers/${test}`, 'utf8')

const input = ((i = 0) => () => __input__[i++])()

// COPY FROM HERE and uncomment this
/*
const fs = require('fs')
const __input__ = fs.readFileSync('input.txt', 'utf8').split('\n')
const input = ((i = 0) => () => __input__[i++])()

//*/
const n = Number(input())
const a = input().split(' ').map(Number)

let p = 0
let s = a.reduce((r, i) => r + i, 0)
let ans = a.reduce((r, i, j) => r + i * j, 0)

for (let i = 0, x = ans; i < n; i++) {
  ans = Math.min(ans, x += (p += a[i]) - (s -= a[i]))
}
console.log(ans)
console.log(+answer)

/*
fs.writeFileSync('output.txt', `${ans}`)
*/
